extensions/provisioning/creates.go

```diff
@@ -120,6 +125,32 @@ func CreateProvisioningCluster(client *rancher.Client, provider Provider, cluste
 
 	machinePools := machinepools.
 		CreateAllMachinePools(machineConfigs, pools, machinePoolResponses, provider.Roles, hostnameTruncation)
+
+	if clustersConfig.CloudProvider == provisioninginput.VsphereCloudProviderName.String() {
+
+		vcenterCredentials := map[string]interface{}{
+			"datacenters": machinePoolConfigs[0].Object["datacenter"],
```

Comment:
This one is always the same right, so I guess my question is out of this PR scope, does it have to index it, or does it have to be in the array?

Reply:
it just has to be in the array